x/blob/keeper/msg_server.go

```diff
@@ -15,3 +19,17 @@ type msgServer struct {
 func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 	return &msgServer{Keeper: keeper}
 }
+
+// PayForBlob consumes gas based on the blob size.
+func (k Keeper) PayForBlob(goCtx context.Context, msg *types.MsgPayForBlob) (*types.MsgPayForBlobResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	gasToConsume := uint64(shares.BlobSharesUsed(int(msg.BlobSize)) * GasPerBlobShare)
+	ctx.GasMeter().ConsumeGas(gasToConsume, payForBlobGasDescriptor)
+
+	ctx.EventManager().EmitEvent(
+		types.NewPayForBlobEvent(sdk.AccAddress(msg.Signer).String(), msg.GetBlobSize()),
+	)
+
+	return &types.MsgPayForBlobResponse{}, nil
+}
```

Comment:
[blocking discussion]
imo if we want to move this here, then we should change the method receiver from `Keeper` to `msgServer` and add more boiler plate to wrap the keeper method.

Reply:
Thanks for flagging. I'll refer to https://github.com/cosmos/cosmos-sdk/blob/main/x/gov/keeper/msg_server.go and can do something similar.